app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -671,13 +669,18 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
         indexPinching = mSmoothIndexThumbDistance < kPinchThreshold;
     }
     delegate.SetPinchFactor(mIndex, pinchFactor);
-    bool triggerButtonPressed = indexPinching && !leftPalmFacesHead && mHasAimState;
+    bool triggerButtonPressed = indexPinching && !palmFacesHead && mHasAimState;
     delegate.SetButtonState(mIndex, ControllerDelegate::BUTTON_TRIGGER,
                             device::kImmersiveButtonTrigger, triggerButtonPressed,
                             triggerButtonPressed, 1.0);
-    if (leftPalmFacesHead) {
+    if (palmFacesHead) {
+#if defined(OCULUSVR)
+        if (mHandeness == Right)
+            return;
+#endif
         delegate.SetButtonState(mIndex, ControllerDelegate::BUTTON_APP, -1, indexPinching, indexPinching, 1.0);
     } else if (mHasAimState) {
+        // TODO (Shouldn't mHasAimState be always true in this case, since palmFacesHead false ?
```

Comment:
You can delete this comment, the answer is no.. Even if the palm is not facing head mHasAimState could be false, because for example we don't have data from the system for the hand joints used to compute the aim.

Reply:
ok, thanks for clarifying it.